### PR TITLE
fix(actions/api): don't duplicate query looking for nonRealtime query

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
This small PR removes the duplicated queries made when planning a trip. There are some minor implications on the performance of the middleware (as mentioned in the removed comment), but I've tested it with a local middleware instance and all features appear to still be working.

### CODE REVIEW

File: lib/actions/api.js [hunk 1]
```diff
@@ -117,11 +117,21 @@ export function routingQuery(searchId = null, updateSearchInReducer = false) {
...
+          .map(({ mode, params }) => ({
+            mode,
+            ...params
+          }))
+          // Remove non-accessible combinations when accessible routing is enabled
+          .filter((combination) => {
+            if (!accessibleRoutingEnabled) return true
+
+            // Explicit check for false is needed to avoid matching
+            // combinations without the pparameter set (undefined)
```
Comment (daniel-heppner-ibigroup) [inline lib/actions/api.js:132]: Fix typo

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Just found a little typo

File: lib/actions/api.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/actions/api.js]: Good catch thanks! Fixed in 47595228122b9d50cefec2bddba18a25e8baa34e

File: lib/actions/api.js [hunk 2]
```diff
@@ -188,50 +206,7 @@ export function routingQuery(searchId = null, updateSearchInReducer = false) {
...
-        )
-          .then(getJsonAndCheckResponse)
-          .then((json) => {
-            // FIXME: This is only performed when ignoring realtimeupdates currently, just
-            // to ensure it is not repeated twice.
-            // FIXME: We should check that the mode combination actually has
-            // realtime (or maybe this is set in the config file) to determine
-            // whether this extra query to OTP is needed.
-            return dispatch(
-              nonRealtimeRoutingResponse({ response: json, searchId })
```
Comment (binh-dam-ibigroup) [inline lib/actions/api.js:226]: Delete in the corresponding files:
- this action (this file is the only place it is used)
- `NON_REALTIME_ROUTING_RESPONSE` from the reducer
- instances of `activeSearch.nonRealtimeResponse`
- instances of `state.otp.useRealtime`
- Other vars and code that gets unused from the above.

Comment (binh-dam-ibigroup) [reply lib/actions/api.js:226]: Would this result in a breaking change commit?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
I think we delete a reducer action and propagate the changes (see comment).

File: lib/actions/api.js [hunk 3]
```diff
@@ -188,50 +206,7 @@ export function routingQuery(searchId = null, updateSearchInReducer = false) {
...
-        //
-        // FIXME: The statement below may no longer apply with future work
-        // involving realtime info embedded in the OTP response.
-        // (That action records an entry again in the middleware.)
-        // For users who opted in to store trip request history,
-        // to avoid recording the same trip request twice in the middleware,
-        // only add the user Authorization token to the request
-        // when querying the non-realtime route.
-        //
-        // The advantage of using non-realtime route is that the middleware will be able to
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/actions/api.js:201]: Yes I think we should do some thinking/investigation about this -- it was my impression that the middleware only took the origin and destination and then made a new trip plan request anyway?

If we do need to make an `ignoreRealtimeUpdates` request then I agree that should only happen from within the user settings.

File: lib/actions/api.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/actions/api.js:226]: See comment above -- I kept these thinking that we would re-use them when we refactor the monitored trips component. I'll add some comments to all these explaining why they're there. 

Something for the backlog though would be to clear out a lot of these reducers. There's plenty of unused ones floating around...

Comment (miles-grant-ibigroup) [reply lib/actions/api.js:227]: I've gone ahead and removed `useRealtime` as it appeared to be a setting that wasn't used anywhere. That led me to have to add typescript in some places and also make a small refactor to the realtime status label. No breaking change though as `useRealtime` wasn't even in the example-config and hasn't been used in years.

Comment (miles-grant-ibigroup) [reply lib/actions/api.js:227]: Using `useRealtime` prior to this commit wouldn't have done anything anyway

File: lib/actions/api.js [hunk 4]
```diff
@@ -188,50 +206,7 @@ export function routingQuery(searchId = null, updateSearchInReducer = false) {
...
-              nonRealtimeRoutingResponse({ response: json, searchId })
-            )
-          })
-          .catch((error) => {
-            console.error(error)
-            // do nothing
-          })
-
-        return Promise.all([realTimeFetch, nonRealtimeFetch])
+        return Promise.all([realTimeFetch])
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/actions/api.js]: Yes! Done in ffcd36c3def760c57acd170d6d993bf0764be177

File: lib/components/mobile/results-header.js [hunk 5]
```diff
@@ -100,18 +100,13 @@ class ResultsHeader extends Component {
 
 const mapStateToProps = (state, ownProps) => {
   const activeSearch = getActiveSearch(state)
-  const {useRealtime} = state.otp
-  const response = !activeSearch
-    ? null
-    : useRealtime ? activeSearch.response : activeSearch.nonRealtimeResponse
+  const response = !activeSearch ? null : activeSearch.response
```
Comment (binh-dam-ibigroup) [inline lib/components/mobile/results-header.js:103]: Would this work?
```suggestion
  const response = activeSearch?.response
```

File: lib/components/mobile/results-screen.js [hunk 6]
```diff
@@ -175,18 +168,14 @@ const mapStateToProps = (state, ownProps) => {
...
+        ? itineraries.length
+        : response.otp.profile.length
+      : null
   }
 }
 
-const mapDispatchToProps = {
-  setUseRealtimeResponse: narrativeActions.setUseRealtimeResponse
-}
+const mapDispatchToProps = {}
```
Comment (binh-dam-ibigroup) [inline lib/components/mobile/results-screen.js:179]: Delete this var and the 2nd arg passed to `connect`.

File: lib/components/viewers/realtime-status-label.tsx [hunk 7]
```diff
@@ -108,19 +118,27 @@ const RealtimeStatusLabel = ({
...
-              <FormattedDuration duration={Math.abs(delay)} />
-            </DelayText>
-          )}
+          minutes={
+            isEarlyOrLate ? (
+              <DelayText>
+                <FormattedDuration duration={Math.abs(delay)} />
+              </DelayText>
+            ) : (
+              <>{null}</>
```
Comment (binh-dam-ibigroup) [inline lib/components/viewers/realtime-status-label.tsx:127]: Just `null` should be enough, no?

File: lib/util/state.js [hunk 8]
```diff
@@ -265,21 +269,18 @@ export const getActiveItineraries = createSelector(
...
-    useRealtime,
     activeFieldTripRequest
   ) => {
-    // set response to use depending on useRealtime
     const response =
       !nonRealtimeResponse && !realtimeResponse
         ? null
-        : useRealtime
+        : realtimeResponse
         ? realtimeResponse
```
Comment (binh-dam-ibigroup) [inline lib/util/state.js:284]: Can this be simplified?

**Review by binh-dam-ibigroup — APPROVED** [review]
A few minor changes but it should be good to go.

File: lib/components/mobile/results-header.js (hunk 5 above)
Comment (miles-grant-ibigroup) [reply lib/components/mobile/results-header.js]: Good call thanks! It's so easy to get lost while refactoring something down like this. Cleaned up in 23ef058a46835bd8ce5710176ca1671ffcf7a586

File: lib/components/mobile/results-screen.js [hunk 9]
```diff
@@ -162,10 +158,7 @@ class MobileResultsScreen extends Component {
 
 const mapStateToProps = (state, ownProps) => {
   const activeSearch = getActiveSearch(state)
-  const {useRealtime} = state.otp
-  const response = !activeSearch
-    ? null
-    : useRealtime ? activeSearch.response : activeSearch.nonRealtimeResponse
+  const response = !activeSearch ? null : activeSearch.response
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply lib/components/mobile/results-screen.js]: Yep! Cleaned up in 23ef058a46835bd8ce5710176ca1671ffcf7a586

File: lib/components/mobile/results-screen.js (hunk 6 above)
Comment (miles-grant-ibigroup) [reply lib/components/mobile/results-screen.js]: Good call thanks! Cleaned up in 23ef058a46835bd8ce5710176ca1671ffcf7a586

File: lib/components/viewers/realtime-status-label.tsx (hunk 7 above)
Comment (miles-grant-ibigroup) [reply lib/components/viewers/realtime-status-label.tsx:127]: If only... Typescript was very upset about the JSX being passed in as a value for FormattedMessage so I had to wrap the null to make it happy

File: lib/util/state.js (hunk 8 above)
Comment (miles-grant-ibigroup) [reply lib/util/state.js]: Yep! in 23ef058a46835bd8ce5710176ca1671ffcf7a586

**Pull request merged**